verify checksums against sha256sum files when available

gfelbing opened the pull request:
Many projects offer a single asset in form of a sha256sum file, e.g. [k9s](https://github.com/derailed/k9s/releases). This PR checks the asset list for such files and verifies a checksum against it.

zyedidia commented:
How is this different than the existing functionality that verifies checksums?

gfelbing commented:
The current functionality only uses single-file checksums, e.g. when there is a `binaryname.sha256` next to `binaryname`.
Many projects offer their checksums in a single sha256 formatted file, e.g. the `checksums.txt` in [k9s](https://github.com/derailed/k9s/releases).
This PR adds the functionality for checking against such files.

hhromic commented:
+1 for this feature.

[GoReleaser](https://github.com/goreleaser/goreleaser) is used by many Go projects nowadays and it generates single checksum files by default as well.
Their own releases page is an example.

gfelbing commented:
@zyedidia I rebased the PR against your latest master and simplified the match for sha256files.
Is there anything you are missing from the PR? 
If you need more examples for the usefulness of the feature: e.g. [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v3.14.0) uses a single checksums file as well.
